### ISSUE TITLE
[0.9.0 staging-1287] Can break shader

### ISSUE THREAD

**Issue by beastyn** [StrangeLoopGames/EcoIssues#14042]
1. Enter the game
2, /tpto another_player

After that something bad happens with textures.
![bandicam 2019-12-06 12-51-33-699](https://user-images.githubusercontent.com/27898520/70307087-2cf26e80-182a-11ea-960c-225ff8fcf217.jpg)
![bandicam 2019-12-06 12-52-19-662](https://user-images.githubusercontent.com/27898520/70307098-3380e600-182a-11ea-91b9-820b247038d1.jpg)

**Comment by mirasrael** [comment]
Log very helpful in such cases.

**Comment by mirasrael** [comment]
@beastyn can you confirm if it still an issue? Close otherwise.

**Comment by SlayksWood** [comment]
Seems fixed.

**Comment by beastyn** [comment]
Sstill here.
Can check with /teleport todark
[Player.log](https://github.com/StrangeLoopGames/EcoIssues/files/4310795/Player.log)

**Comment by SlayksWood** [comment]
#15129
> if you move too fast and enter onto not loaded chunk - your shaders is broken until game restart
> easy to see - fly to not loaded chunks.
> 
> happens with server lags, driving vehicle at asphalt, etc.
> 
> ![20200314150559_1](https://user-images.githubusercontent.com/4980243/76681594-af4ab580-6605-11ea-885b-b1c7eb67b516.jpg)
> ![20200314150603_1](https://user-images.githubusercontent.com/4980243/76681595-b07be280-6605-11ea-8b00-4b0e863ff725.jpg)
> ![20200314150606_1](https://user-images.githubusercontent.com/4980243/76681596-b1ad0f80-6605-11ea-821e-a6c78b7b6516.jpg)
> ![20200314150612_1](https://user-images.githubusercontent.com/4980243/76681597-b245a600-6605-11ea-8ee2-4e1b4e92a705.jpg)
> ![20200314150626_1](https://user-images.githubusercontent.com/4980243/76681599-b2de3c80-6605-11ea-9fc7-1c5b89909f92.jpg)

**Comment by elfl0rd** [comment]
if player log still needed, ping me, i attach it.

**Comment by mirasrael** [comment]
@elfl0rd yes, please attach the log

**Comment by elfl0rd** [comment]
[Player.zip](https://github.com/StrangeLoopGames/EcoIssues/files/4338345/Player.zip)

**Comment by mirasrael** [comment]
Issue happens because of `unity_IndirectSpecColor` after teleport (sometimes) becomes invalid.
It should be {r: 0.036938496, g: 0.036938496, b: 0.036938496, a: 0.3}, but for some reason (Unity bug?) becomes {r: NaN, g: NaN, b: NaN, a: 0.3}.

**Comment by mirasrael** [comment]
Seems like Unity bug and we wasn't able to reproduce it in 2019.3. Put it on hold for now.

**Comment by WeaselDog** [comment]
This happened to me during the live stream today using Steam build.

**Comment by SlayksWood** [comment]
> Sometimes some Parts of the Texture of Objects are Black
> ![grafik](https://user-images.githubusercontent.com/37456949/82161705-37b43380-989f-11ea-9c5b-4453c5d84e4e.png)
> 
> When you change now the Graphic Setting to Other value and then Back its normal again
> [Player.log](https://github.com/StrangeLoopGames/EcoIssues/files/4642499/Player.log)
> ![grafik](https://user-images.githubusercontent.com/37456949/82183087-a0be9a00-98e5-11ea-8faa-393d1d7a9182.png)

**Comment by elfl0rd** [comment]
9.1? it happens in normal gameplay proces while fast movement...

**Comment by mirasrael** [comment]
We just have to wait Unity fix.

**Comment by SlayksWood** [comment]
> ![20200726142905_1](https://user-images.githubusercontent.com/68478692/88486545-770c9980-cf4c-11ea-9468-eab4034ebe91.jpg)
> It looks like something is causing the inside of my house to be black. I even removed a part of the roof and it was still black inside my house. Everything returned to normal after restart. This happened after I walked away from my house and came back.
> **After Restart:**
> ![20200726143641_1](https://user-images.githubusercontent.com/68478692/88486676-758fa100-cf4d-11ea-8dc7-60af1a4a854a.jpg)
> Steps to reproduce:
> 
> 1. Build a room with windows.
> 2. Fly into unloaded chunks
> 3. Wait for the unconnected icon to show up(red icon in bottom right corner)
> 4. Fly back to house while the icon is still there.
>    When you get back the room will be entirely black.
>    **The bug can still happen when not flying this is just a way to reproduce, originally it happened after I came back from hunting a deer**

**Comment by mirasrael** [comment]
@redwyre can you please report this issue to Unity?

**Comment by EcoBuilder13** [comment]
@SlayksWood 
CPU: Intel Core i7-8750H
Ram: 16GB 2667 MHz
GPU: Nvidia GTX 1060 (6GBs of VRAM)
Storage Drive: PCIE Gen 3 NVME M.2 SSD
**If you need anything else let me know.**

**Comment by GreaseMonk** [comment]
This seems to happen after playing after a while, it seems

**Comment by GreaseMonk** [comment]
CPU Specs:
Processor - Intel(R) Core(TM) i7-4790K CPU @ 4.00GHz, 4001 Mhz, 4 Core(s), 8 Logical Processor(s)
System Manufacturer - Gigabyte Technology Co., Ltd.
System Model	- Z97M-D3H
GPU - NVIDIA GeForce GTX 970 - Driver version 446.16


OS Name	Microsoft Windows 10 Pro
Version	10.0.18363 Build 18363

**Comment by GreaseMonk** [comment]
Havent seen this happen anymore since staging-1756 I think. 
Can anyone else confirm it's been resolved ?

**Comment by MishkaFur** [comment]
@GreaseMonk Ive had it happen on the WT prep server, but it seems to be most common if you TP into an area of shade, if its daylight its fine.

**Comment by SlayksWood** [comment]
> **Version:** 0.9.0.2 beta release-71
> 
> **Steps to Reproduce:**
> 
> 1. Start the Game
> 2. Wait for the first night
> 3. Since then all effects and windows are black
> 
> **Expected behavior:**
> Normal, transparent. Shaderbug i think
> 
> **Actual behavior:**
> Black, shader incorrect
> 
> **Do you have mods installed? Does the issue happen when no mods are installed?:**
> No and no
> 
> **System:**
> Intel i7-4790
> 24 GB (DDR3) RAM
> AMD R9 Fury Nitro (Sapphire) 4 GB VRAM (HBM) [Radeon Software Driver 20.9.1]
> Windows 10 Pro
> 
> **Please provide log files. Instruction is here https://github.com/StrangeLoopGames/EcoIssues/wiki/Getting-the-Eco-game-log-file
> [Player.log](https://github.com/StrangeLoopGames/EcoIssues/files/5254212/Player.log)
> [Player-prev.log](https://github.com/StrangeLoopGames/EcoIssues/files/5254213/Player-prev.log)
> [output_log.txt](https://github.com/StrangeLoopGames/EcoIssues/files/5254216/output_log.txt)
> ![Eco_2020_09_21_10_15_26_269](https://user-images.githubusercontent.com/71633422/93747092-7b74cc00-fbf6-11ea-9122-32c37f78674b.png)

**Comment by GreaseMonk** [comment]
I haven't had this since I just disabled shadows, it seems to appear pretty quickly when setting a higher shadows quality.

**Comment by johnkslg** [comment]
This is passed to Unity and waiting on bug fix from them.

**Comment by ptitducanh** [comment]
I sent a bug report to Unity.
https://fogbugz.unity3d.com/default.asp?1285723_74gqqml451059thq

**Comment by MishkaFur** [comment]
Still exists in 9.3 1971 :D

if disabling basically all the tickbox stuff I do not get broken shaders though.

![image](https://user-images.githubusercontent.com/61889138/113941784-aec82280-97f7-11eb-8f0b-e3d84cca27f1.png)

![image](https://user-images.githubusercontent.com/61889138/113941801-ba1b4e00-97f7-11eb-9a25-19b58871ab7b.png)
